client/extensions/woocommerce/app/order/order-customer/dialog.js

```diff
@@ -75,7 +76,11 @@ class CustomerAddressDialog extends Component {
 	state = {};
 
 	componentDidMount() {
+		const { siteId } = this.props;
 		this.initializeState();
+		if ( siteId && ! this.props.areLocationsLoaded ) {
```

Comment:
I liked your `maybeFetchLocations` in the component above, could be used here also.